rad/monadic/issues.rad

```diff
@@ -0,0 +1,142 @@
+  "Create an open issue with no comments or labels."
+  (fn [tit bod]
+    {:title      tit
+     :body       bod
+     :labels     []
+     :state      :open
+     :comments   []
+     :created-at (now!)}))
+
+(def simple-comment
+  "Create a comment for an issue."
+  (fn [issue-number bod]
+    {:body bod
+     :issue-number issue-number
+     :created-at (now!)}))
+
+(def create-issue!
+  "Create a new remote issue with the keys in `my-keys.rad`."
+  (fn [i]
+    (send! (lookup :url (read-ref issues-chain))
+           [(list 'create-issue (sign-entity! i))])))
+
+(def add-comment!
+  "Create a remote comment with the keys in `my-keys.rad`."
+  (fn [c]
+    (send! (lookup :url (read-ref issues-chain))
+           [(list 'add-comment (sign-entity! c))])))
+
+(def simple-create-issue!
+  "Create a remote issue with sensible defaults."
+  (fn [tit bod]
+    (create-issue! (simple-issue tit bod))))
+
+(def simple-add-comment!
+  "Create a remote comment."
+  (fn [in b]
+    (add-comment! (simple-comment in b))))
+
+(def list-issues
+  "Return the full map of issues."
+  (fn []
+    (lookup :result (eval-in-chain '(list-issues) (read-ref issues-chain)))))
+
+(def fetch
+  "Update the locally cached state of the chain."
+  (fn []
+    (update-chain-ref issues-chain)))
+
+(def import-github
+  "Imports the github issues stored in file `githubissues.rad`. To create this
+  file use the `radicle-github-issues` executable. Will add a note to all
+  `:body`s that this is an imported entity."
+  (fn []
+    (def note-imported
+      (fn [x]
+        (insert
+         :body
+         (string-append
+          (lookup :body x)
+          "\n\n*Imported from GitHub.* Original author: `"
+          (lookup :github-username x)
+          "`.")
+         x)))
+    (def note-comments
+      (fn [i] (over (@ :comments) (fn [cs] (map note-imported cs)) i)))
+    (def issues (read (read-file! "githubissues.rad")))
+    (def imported
+      (map
+       (fn [i] (note-imported (note-comments i)))
+       issues))
+    (map create-issue! imported)))
+
+(def short
+  "Short render of an issue."
+  (fn [i]
+    (string-append
+     (show (lookup :number i))
+     ": "
+     (lookup :title i))))
+
+(def list-last
+  "Get the last `n` issues."
+  (fn [n]
+    (def is (sort-by (fn [i] (- 0 (lookup :number i))) (values (list-issues))))
+    (take n is)))
+
+(def get-by-id
+  (fn [id]
+    (lookup id (list-issues))))
+
+(:test
+  "creating, manipulating and querying the issue chain"
+  [:setup
+     (do
+       (def pk (gen-key-pair! (default-ecc-curve)))
+       (def chain-name (uuid!))
+       (def chain-ref (create-issues-chain! chain-name))
+       (new-issue! chain-ref pk "Ah!" "if thou issueless shall hap to die")
+       (update-chain-ref chain-ref)
+       (def issues (values (list-issues chain-ref))))
+   ]
+   [ (length issues) ==> 1 ]
+   [ (lookup :title (nth 0 issues)) ==> "Ah!" ]
+   [ (lookup :body (nth 0 issues)) ==> "if thou issueless shall hap to die" ]
```

Comment:
Definitely needs more tests

Reply:
Added a test but it doesn't work, whenever I put more than one test it fails.